src/picongpu/include/particles/gasProfiles/FreeFormulaImpl.hpp

```diff
@@ -51,7 +51,8 @@ struct FreeFormulaImpl : public T_ParamClass
      */
     HDINLINE float_X operator()(const DataSpace<simDim>& totalCellOffset)
     {
-        const floatD_64 cellSize_SI( precisionCast<float_64>(cellSize) * UNIT_LENGTH );
+        const float_64 unitLength = UNIT_LENGTH; // workaround to use UNIT_LENGTH on device 
```

Comment:
one thing is, that it generally uses the constructor instead of an "=" which is an assignment. an other point is, if you look at the lines above and below, than a patch should generally fit the style of the surrounding code (a patch should not be visible after it is applied).